app/scripts/directives/unbindService.js

```diff
@@ -26,11 +26,13 @@
     var serviceInstanceDisplayName = $filter('serviceInstanceDisplayName');
 
     var unbindService = function() {
+      ctrl.deletedBinding = _.first(_.filter(ctrl.bindings, {metadata: {name: ctrl.selectedBinding}}));
+      ctrl.appsForDeletedBinding = angular.copy(ctrl.appsForBinding(ctrl.selectedBinding));
```

Comment:
You still shouldn't need a copy because you have a reference to the old array, which is not mutated by the overview controller. The entire map is replaced, so the previous array is unchanged. It should work without the copy.